### ISSUE TITLE
Windows - full rebuild on every init due to "changes" in Emacs version

### ISSUE THREAD

**Issue by sivertsenstian** [raxod502/straight.el#276]
Not sure why this is happening, my current emacs version stays the same when I print it:
![image](https://user-images.githubusercontent.com/1849292/40047442-c5cfeee6-582f-11e8-85cf-6ad468b339fe.png)

Not sure how I check what straight gets as `last-emacs-version ` when it compares the versions:
https://github.com/raxod502/straight.el/blob/ac4cfb80fc0ce59ee3a2e8d6aebf100731a44e0a/straight.el#L2528

In messages I get (on every startup)

>Rebuilding all packages due to change in Emacs version
>Building straight...done
>Building use-package...
>Building use-package → Building bind-key...done
>Building use-package...done
>Building evil...
>Building evil → Building undo-tree...done
>Building evil → Building goto-chg...done
>Building evil...done
>Building evil-leader...done
>Building evil-escape...done
>Building evil-exchange...done
>Building evil-matchit...done
>Building evil-snipe...done
.... etc.

> init completed in 114424.79ms 😢 

I tried adding
```
(setq straight-cache-autoloads t
      straight-check-for-modifications 'never)
```
To the top of my init.el (after the straight bootstrap snippet), to try and work around it, but did not seem to help.

Using emacs from https://ftp.gnu.org/gnu/emacs/windows  
the latest .zip

Not sure if this is an issue with my emacs. straight or my config.. If you have any idea why this is happening or what I can do to investigate further I would appreciate it.

**Comment by janoc** [comment]
I see this same problem in windows too. No such issues on Mac or Linux.

**Comment by janoc** [comment]
I have just done a test and this is what I get in my messages window:

Rebuilding all packages due to change in Emacs version

And the last-emacs-version is (sans quotes):

"GNU Emacs 26.0.90 (build 1, x86_64-w64-mingw32)
 of 2017-11-30"

I.e. there is CRLF between the two lines.

And here is my build-cache.el:
https://pastebin.com/AsJg9uEZ

**Comment by raxod502** [comment]
Are you sure they are the same? If they were, then presumably the `equal` check would not fail. I think that one of them has a CRLF, and the other only has an LF.

I looked at your build cache, and indeed it has the CRLF as you said. I guess we need to decide whether the build cache should have a CRLF or an LF, and change the other string that is compared accordingly.

**Comment by janoc** [comment]
I am sure it was CRLF and not just LF because the Messages window was showing the ^M character there. 

You could probably normalize the strings too - there is no very good reason why the string should have newline characters in it.

**Comment by raxod502** [comment]
> You could probably normalize the strings too - there is no very good reason why the string should have newline characters in it.

True. It would still be nice if we weren't reading and writing the cache file in an asymmetric way, but I wouldn't object to a normalization step.

**Comment by janoc** [comment]
Hello, I have made a quick fix here #277

**Comment by raxod502** [comment]
I guess that solves the problem.
